test/credential-test.js

```diff
@@ -156,6 +156,61 @@ test('verify with empty password', function(t) {
 
 });
 
+test('constantEquals', function (t) {
+  var ctc = require('../constantTimeCompare');
+
+  function timed_compare(a, b) {
+    var start = process.hrtime();
+    ctc(a, b);
+    return process.hrtime(start)[1];
+  }
+  var i,
+      iterations = 5000,
```

Comment:
I'd really like one or more of the JavaScript engine optimization experts to review this and make sure that what we're doing here is enough to prime the optimizations.